eng/pipelines/templates/jobs/archetype-go-release.yml

```
@@ -29,6 +29,13 @@ stages:
                     parameters:
                       PackageName: 'sdk/${{ parameters.ServiceDirectory }}'
                       ForRelease: true
+                  - task: PowerShell@2
+                    displayName: 'Require no replace directives in go.mod file'
```

Nit
```suggestion
                    displayName: 'Verify go.mod file does NOT contain replace directives'
```

---

I would also suggest below because it will probably show up in the truncated display name UI as "Verify go.mod file"

```suggestion
                    displayName: 'Verify no replace directives in go.mod file'
```